.github/workflows/twister.yaml

```diff
@@ -213,6 +213,17 @@ jobs:
           ccache -p
           ccache -z -s -vv
 
+      - name: Update BabbleSim to manifest revision
+        run: |
+          export BSIM_VERSION=$( west list bsim -f {revision} )
+          echo "Manifest points to bsim sha $BSIM_VERSION"
+          cd /opt/bsim_west/bsim
+          git fetch -n origin ${BSIM_VERSION}
+          git config --global advice.detachedHead false
+          git checkout ${BSIM_VERSION}
+          west update
```

Comment:
hey I'm a bit confused, shouldn't the existin `west update` in `Environment Setup` step take care of this already?

Reply:
For the Babblesim simulator we do not get it (and rebuild it) each time, but instead we keep a "preinstalled" version in the docker image in /opt. Apart from historic reasons, it saves a few minutes.
The simulator changes very rarely and an incremental fetch and build is ~1second vs a full build (several minutes).

Reply:
oh because bsim is in the `babblesim` troup and the worfklow only does `ci` and `optional`? Ok but then is this `west update` command needed? I should not do anything since it's already been run before.

Reply:
oookay.. I'll knock myself out :-)

Reply:
It is a fair question :). Zephyr keeps nested in its manifest the bsim project manifest as a disabled group. Users are recommended to use that by just enabling that group.
But for CI we don't use it from there.
All we do here is check what version the Zephyr manifest is pointing at (store it in BSIM_VERSION); then change directory to /opt/bsim_west/bsim, where a standalone bsim can be found (without anything from Zephyr), and update that version to whatever Zephyr's manifest points to. (here `west` is used as a repository management tool, just like google's `repo` fully detached from Zephyr itself)
Normally BSIM_VERSION will just be the one prebuilt in the docker image, but sometimes (during updates) the manifest may be ahead of the docker image, or we may be running in a newer docker image with an older version of Zephyr for whatever reason.
With this minor trick, we both save time not fetching and rebuilding bsim in each CI runner, and ensure we run with the version pointed by the Zephyr manifest no matter the CI image.

Reply:
Yeah got it now, I just got confused by that west command being there did not realize was another full checkout there, makes sense.

Then I think you could also pass `-o=--depth=1 -n` to make it download even less but it's likely milliseconds of difference in this case.